src/components/DataExplorerApp/VizSettingsForm/index.tsx

```diff
@@ -39,6 +33,21 @@ export function VizSettingsForm({
     labelFn: getProp("displayName"),
   });
 
+  // helper to seed next config with cached XY
+  function seedXY<T extends VizConfig>(next: T): T {
+    const xy = vizConfig.cachedXY;
+    if (!xy) return next;
+    // only charts have settings to merge into; table just carries the cache
+    if (next.type === "bar" || next.type === "line") {
```

Comment:
Use `match()` from `ts-pattern` so we can have a type-safe switch case which will tell us if we ever aren't checking all viz types